### DISCUSSION TITLE
Version release

### DISCUSSION THREAD

**Discussion by hovind** [MikePopoloski/slang#1171] [Q&A] [answered]
Hi, thanks for the fantastic library!

I'm trying to update `circt` to depend on a newer `slang` release over at https://github.com/llvm/circt/pull/7792. A simple way I found to pass the `mlir` buffers to the `slang` driver is https://github.com/MikePopoloski/slang/pull/1164, which is not yet (as far as I can tell) part of a release.

Is there any chance of getting a release with this feature in the not too distant future? It would be greatly appreciated.

**Comment by MikePopoloski** [comment, marked as answer]
7.0 was fairly recent so I wasn't planning on doing the 8.0 release for a few more months. There's no intrinsic reason it can't be done sooner though if it's really needed. It's probably worth making sure there aren't other changes you need too before I do so.

**Reply by hovind** [reply]
Thanks for the quick response. That sounds great, I'll possibly inquire further when the pull request over in the `circt` repository is in a more polished state.

There's one more thing that came up when updating `slang` that I would like to ask about. I attempted to use a raw [SourceBuffer](https://github.com/MikePopoloski/slang/blob/41ee213adc869b89764522c197be91854b1ab52b/include/slang/text/SourceLocation.h#L208) to avoid having to copy the content of the `llvm::MemoryBuffer`, but ran into [the lexer](https://github.com/MikePopoloski/slang/blob/master/source/parsing/Lexer.cpp#L49) expecting source buffers to be null terminated.

This copying is not a big deal, and changing this behavior may be a ton of work, but I just wanted to ask if you had any thoughts on supporting source buffers that aren't null terminated?

**Reply by MikePopoloski** [reply]
Yeah, the Lexer doesn't do bounds checking except when encountering a null byte in the text; it would be pretty difficult to change. I think clang / LLVM do the same though; can you not have CIRCT / MLIR load the buffers with a trailing null appended? I think there are built-in ways in LLVM of doing that so perhaps it's not hard.

As a separate issue, you can't really use an externally provided buffer with slang right now anyway. Things like SourceLocations expect that the buffer is known to the SourceManager; I'm not sure exactly what will break if you just magic up a SourceBuffer and pass it through the system but it's not designed to work that way. It definitely makes sense to add an API to the SourceManager though that can adopt an external buffer without the copy though, it just hasn't been at the top of the priority list.
